facade/src/main/java/org/jboss/pnc/facade/providers/ProductMilestoneProviderImpl.java

```diff
@@ -275,6 +281,34 @@ public void cancelMilestoneCloseProcess(String id) throws RepositoryViolationExc
         return new Page<>(pageIndex, pageSize, milestoneIds.size(), milestones);
     }
 
+    @Override
+    public ValidationResponse validateVersion(String productVersionId, String version) {
+        boolean matches = Pattern.matches(Patterns.PRODUCT_MILESTONE_VERSION, version);
+        ValidationResponse.Builder builder = ValidationResponse.builder().isValid(matches);
+        if (!matches) {
+            return builder.errorType(FORMAT)
+                    .hints(
+                            Lists.newArrayList(
```

Comment:
I would prefer Collections.singletonList() instead of using guava library

Reply:
Sure